x/nexus/keeper/chain.go

```diff
@@ -128,57 +129,66 @@ func (k Keeper) IsChainActivated(ctx sdk.Context, chain exported.Chain) bool {
 	return chainState.Activated
 }
 
-// GetChainMaintainerStates returns the maintainer states of the given chain
-func (k Keeper) GetChainMaintainerStates(ctx sdk.Context, chain exported.Chain) []types.MaintainerState {
-	chainState, ok := k.getChainState(ctx, chain)
+// GetChainMaintainerState returns the maintainer state of the given chain and address
+func (k Keeper) GetChainMaintainerState(ctx sdk.Context, chain exported.Chain, address sdk.ValAddress) (exported.MaintainerState, bool) {
+	ms, ok := k.getChainMaintainerState(ctx, chain.Name, address)
 	if !ok {
-		return []types.MaintainerState{}
+		return nil, false
 	}
 
-	return chainState.MaintainerStates
+	return &ms, true
+}
+
+// GetChainMaintainerStates returns the maintainer states of the given chain
+func (k Keeper) GetChainMaintainerStates(ctx sdk.Context, chain exported.Chain) []exported.MaintainerState {
+	return slices.Map(k.getChainMaintainerStates(ctx, chain.Name), func(ms types.MaintainerState) exported.MaintainerState {
+		return &ms
+	})
+}
+
+// SetChainMaintainerState sets the given chain's maintainer state
+func (k Keeper) SetChainMaintainerState(ctx sdk.Context, maintainerState exported.MaintainerState) error {
+	ms := maintainerState.(*types.MaintainerState)
```

Comment:
Panic on type cast error?

Reply:
Yes cuz passing something in that's not a MaintainerState but happens to implement the interface is clearly a developer error instead of user error.